src/object/obj_verify.c

```diff
@@ -211,16 +211,18 @@ dc_obj_verify_parse_dkey(struct dc_obj_verify_args *dova, daos_obj_id_t oid,
 	dkey.iov_buf_len = dova->kds[idx].kd_key_len;
 	dkey.iov_len = dova->kds[idx].kd_key_len;
 
-	if (gen == cursor->gen)
+	if (gen == cursor->gen &&
+	    (cursor->dkey.iov_buf == NULL ||
+	     !daos_key_match(&cursor->dkey,&dkey)))
```

Comment:
(style)  space required after that ',' (ctx:VxO)
(style)  space required before that '&' (ctx:OxV)